File: shanny-blog-java/blog-pojo/src/main/java/com/shanny/entity/Category.java
package com.shanny.entity;

import com.shanny.enums.ArticleEnum;
import com.shanny.enums.CategoryEnum;
import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import lombok.Data;
import org.springframework.format.annotation.DateTimeFormat;

import java.time.LocalDateTime;

@Data
@Entity
@Table(name = "categories")
public class Category {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @NotNull
    @Column(length = 10, nullable = false)
    private String name;

    @NotNull
    @Column(length = 10, nullable = false)
    private String nameEn;

    @NotNull
    @Column(nullable = false)
    @Enumerated(EnumType.STRING)
    private CategoryEnum.CategoryType type;

    @NotNull
    @Column(nullable = false)
    private int sort = 0;
}

File: shanny-blog-net/blog-pojo/Entities/Category.cs
using blog_common.Enums;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace blog_pojo.Entities
{
    [Table("categories")]
    public class Category
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        [Column("id")]
        public long Id { get; set; }

        [MaxLength(10)]
        [Required]
        [Column("name")]
        public string Name { get; set; } = string.Empty;

        [MaxLength(10)]
        [Required]
        [Column("nameEn")]
        public string NameEn { get; set; } = string.Empty;

        [Required]
        [Column("type")]
        public CategoryType Type { get; set; }

        [Required]
        [Column("sort")]
        public int Sort { get; set; } = 0;
    }
}
